feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts

```diff
@@ -42,22 +42,51 @@ export class CheckoutPaymentTypeComponent {
   paymentTypes$: Observable<PaymentType[]> =
     this.checkoutPaymentTypeFacade.getPaymentTypes();
 
-  typeSelected$: Observable<PaymentType> = this.checkoutPaymentTypeFacade
-    .getSelectedPaymentTypeState()
-    .pipe(
+  typeSelected$: Observable<PaymentType> = combineLatest([
+    this.checkoutPaymentTypeFacade.getSelectedPaymentTypeState().pipe(
       filter((state) => !state.loading),
-      map((state) => state.data),
-      filter(isNotUndefined),
-      distinctUntilChanged(),
-      tap((selected) => {
-        this.typeSelected = selected?.code;
-        this.checkoutStepService.resetSteps();
-        this.checkoutStepService.disableEnableStep(
-          CheckoutStepType.PAYMENT_DETAILS,
-          selected?.code === B2BPaymentTypeEnum.ACCOUNT_PAYMENT
-        );
-      })
-    );
+      map((state) => state.data)
+    ),
+    this.paymentTypes$,
+  ]).pipe(
+    map(([prevSelected, availablePaymentTypes]) => {
+      if (prevSelected) {
```

Comment:
This can be combined into one `if` statement I guess:

```ts
if (
        selectedPaymentType &&
        availablePaymentTypes.find(
          (availablePaymentType) =>
            availablePaymentType.code === selectedPaymentType?.code
        )
      ) {
        return selectedPaymentType;
      }
```

Reply:
Yes, I combined them. Thank you